File: 백준/Gold/1806. 부분합/부분합.py
import sys
input = lambda: sys.stdin.readline().rstrip()
print = lambda *args, sep=" ", end="\n": sys.stdout.write(sep.join(map(str, args)) + end)

from math import inf

def main():
    n, s = map(int, input().split())
    a = [*map(int, input().split())]
    ans = inf
    l = suma = 0
    for r in range(n):
        suma += a[r]
        while suma >= s:
            ans = min(ans, r-l+1)
            suma -= a[l]
            l += 1
    print(0 if ans == inf else ans)
    return
if __name__ == "__main__":
    main()

File: 백준/Gold/1806. 부분합/부분합.cc
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

int main() {
    cin.tie(0); cout.tie(0); ios::sync_with_stdio(false);
    int n, s; cin >> n >> s;
    vector<int> a(n);
    for(int i = 0; i < n; i++) cin >> a[i];
    int ans = INT_MAX, l = 0, suma = 0;
    for(int r = 0; r < n; r++){
        suma += a[r];
        while(suma >= s){
            ans = min(ans, r-l+1);
            suma -= a[l];
            l++;
        }
    }
    cout << (ans == INT_MAX ? 0 : ans);
    return 0;
}